apps/server/src/services/rundown-service/delayUtils.ts

```diff
@@ -109,27 +105,69 @@ export function apply(eventId: string, rundown: OntimeRundown): OntimeRundown {
     throw new Error('Given event ID is not a delay');
   }
 
-  const updatedRundown = [...rundown];
-  const delayValue = delayEvent.duration;
-
-  if (delayValue === 0 || delayIndex === rundown.length - 1) {
-    // nothing to apply
-    return updatedRundown;
+  // if the delay is empty, or the last element, we can just delete it
+  if (delayEvent.duration === 0 || delayIndex === rundown.length - 1) {
+    return deleteAtIndex(delayIndex, rundown);
   }
 
-  for (let i = delayIndex + 1; i < rundown.length; i++) {
-    const currentEvent = updatedRundown[i];
+  /**
+   * We apply the delay to the rundown
+   * This logic is mostly in sync with rundownCache.generate
+   */
+  const updatedRundown = structuredClone(rundown);
+  let delayValue = delayEvent.duration;
+  let lastEntry: OntimeEvent | null = null;
+
+  for (let i = delayIndex + 1; i < updatedRundown.length; i++) {
+    const currentEntry = updatedRundown[i];
+
+    // we need to remove the link in the first event to maintain the gap
+    let shouldUnlink = i === delayIndex + 1;
+
+    // we don't do operation on other event types
+    if (!isOntimeEvent(currentEntry)) {
+      continue;
+    }
+
```

Comment:
something like this is needed

```suggestion
  let lastEntry: OntimeEvent | null = null;
  let nonEventAccumilator = 0;

  for (let i = delayIndex + 1; i < updatedRundown.length; i++) {
    const currentEntry = updatedRundown[i];

    // we don't do operation on other event types
    if (!isOntimeEvent(currentEntry)) {
      nonEventAccumilator++;
      continue;
    }
    
    // we need to remove the link in the first event to maintain the gap
    let shouldUnlink = i === delayIndex + 1 + nonEventAccumilator;

```